### ISSUE TITLE
[Sidebars]Maximized windows hide behind taskbar

### ISSUE THREAD

**Issue by akman15** [raiguard/ModernGadgets#227]
On 1.7.On my second monitor i have a sidebar on the right with gadgets and on the left i have the taskbar.When i maximize a window,part of it is behind the taskbar.
![2021-01-30 13_25_34-Greenshot](https://user-images.githubusercontent.com/5766520/106355107-aab2ec80-62fe-11eb-99e4-c66498db4136.png)

**Comment by raiguard** [comment]
Man, the whole sidebars thing is barely functional. I'm surprised there aren't more major issues with it.

I'll basically have to re-write the logic for it to fix this, so it'll take a bit.

**Comment by errornosignal** [comment]
Yeah, same here. I have my gadget panel slid all the way to the right-hand-side of my right-most monitor. When I first boot up, maximizing a window only fills the desktop space not taken up by the sidebar. But after a while (and now that I think of it, it is probably after waking from a sleep state) maximizing a window on that monitor causes the window to take up the entire monitor, covering the sidebar completely. It doesn't bother me at all really, since a quick WIN+[arrow] fixes it easily. Just adding some info that might help lead you to solving it.

**Comment by raiguard** [comment]
So, this problem is turning out to be a giant PITA. Rainmeter's work area functionality is barely functional, and the sidebars are also a barely-functional hack. I honestly don't know if I'll be able to fix your issue.

**Comment by Jimmm** [comment]
I use WorkspaceUtilities:
https://sourceforge.net/projects/workspaceutilities/
It has "Space reserver" option.
No problems so far.

**Comment by raiguard** [comment]
I tried WorkspaceUtilities but it instantly rendered my Windows unusable. Though I am running Windows in a VM so that could be contributing...

I found another tool called [MaxMax](http://mthings.net/) that does the trick pretty well. It's fairly old, but it works well.

I'll probably discourage use of the sidebars or remove them entirely. Rainmeter's work area features just don't cut it, and external tools can do the job much better.
